presto-main-base/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -327,4 +332,81 @@ public boolean validate(Identifier baseTableColumn, Map<Expression, Identifier>
             return baseToViewColumnMap.containsKey(new Cast(new FunctionCall(APPROX_SET, ImmutableList.of(baseTableColumn)), VARBINARY));
         }
     }
+
+    /**
+     * Generate WHERE predicates for missing partitions from MaterializedDataPredicates.
+     * Used for auto-refresh of materialized views without explicit WHERE clause.
+     */
+    public static Map<SchemaTableName, Expression> generatePredicatesForMissingPartitions(
+            Map<SchemaTableName, MaterializedViewStatus.MaterializedDataPredicates> missingPartitionsPerTable,
+            Metadata metadata)
+    {
+        Map<SchemaTableName, Expression> predicates = new HashMap<>();
+
+        for (Map.Entry<SchemaTableName, MaterializedViewStatus.MaterializedDataPredicates> entry :
+                missingPartitionsPerTable.entrySet()) {
+            SchemaTableName tableName = entry.getKey();
+            MaterializedViewStatus.MaterializedDataPredicates missingPartitions = entry.getValue();
+
+            Expression predicate = convertMaterializedDataPredicatesToExpression(missingPartitions, metadata);
+
+            predicates.put(tableName, predicate);
+        }
+
+        return predicates;
+    }
+
+    /**
+     * Convert MaterializedDataPredicates to a SQL Expression tree.
+     * Builds an OR expression of partition predicates, where each partition is an AND expression of column filters.
+     */
+    public static Expression convertMaterializedDataPredicatesToExpression(
+            MaterializedViewStatus.MaterializedDataPredicates predicates,
+            Metadata metadata)
+    {
+        List<String> columnNames = predicates.getColumnNames();
+        List<TupleDomain<String>> predicateDisjuncts = predicates.getPredicateDisjuncts();
+
+        ExpressionDomainTranslator translator = new ExpressionDomainTranslator(
+                new LiteralEncoder(metadata.getBlockEncodingSerde()));
+
+        List<Expression> disjuncts = new ArrayList<>();
+
+        for (TupleDomain<String> tupleDomain : predicateDisjuncts) {
+            if (tupleDomain.isAll() || tupleDomain.isNone()) {
```

Comment:
Is it correct to continue if a domain `isAll`?  This means, all data is stale--shouldn't we return if we see this condition, instead of breaking?

Reply:
If this can't happen in practice, would it make sense to add a `checkState` to ensure that the TD isn't all?  It both would fail fast if that presumption no longer holds, and also signals to readers of this code that this condition can't happen in practice.